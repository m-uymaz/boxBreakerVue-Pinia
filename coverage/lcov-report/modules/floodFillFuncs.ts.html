
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for modules/floodFillFuncs.ts</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../index.html">All files</a> / <a href="index.html">modules</a> floodFillFuncs.ts</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">93.75% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>45/48</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">90% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>18/20</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">87.5% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>7/8</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">94.87% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>37/39</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a>
<a name='L60'></a><a href='#L60'>60</a>
<a name='L61'></a><a href='#L61'>61</a>
<a name='L62'></a><a href='#L62'>62</a>
<a name='L63'></a><a href='#L63'>63</a>
<a name='L64'></a><a href='#L64'>64</a>
<a name='L65'></a><a href='#L65'>65</a>
<a name='L66'></a><a href='#L66'>66</a>
<a name='L67'></a><a href='#L67'>67</a>
<a name='L68'></a><a href='#L68'>68</a>
<a name='L69'></a><a href='#L69'>69</a>
<a name='L70'></a><a href='#L70'>70</a>
<a name='L71'></a><a href='#L71'>71</a>
<a name='L72'></a><a href='#L72'>72</a>
<a name='L73'></a><a href='#L73'>73</a>
<a name='L74'></a><a href='#L74'>74</a>
<a name='L75'></a><a href='#L75'>75</a>
<a name='L76'></a><a href='#L76'>76</a>
<a name='L77'></a><a href='#L77'>77</a>
<a name='L78'></a><a href='#L78'>78</a>
<a name='L79'></a><a href='#L79'>79</a>
<a name='L80'></a><a href='#L80'>80</a>
<a name='L81'></a><a href='#L81'>81</a>
<a name='L82'></a><a href='#L82'>82</a>
<a name='L83'></a><a href='#L83'>83</a>
<a name='L84'></a><a href='#L84'>84</a>
<a name='L85'></a><a href='#L85'>85</a>
<a name='L86'></a><a href='#L86'>86</a>
<a name='L87'></a><a href='#L87'>87</a>
<a name='L88'></a><a href='#L88'>88</a>
<a name='L89'></a><a href='#L89'>89</a>
<a name='L90'></a><a href='#L90'>90</a>
<a name='L91'></a><a href='#L91'>91</a>
<a name='L92'></a><a href='#L92'>92</a>
<a name='L93'></a><a href='#L93'>93</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">268x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">60x</span>
<span class="cline-any cline-yes">60x</span>
<span class="cline-any cline-yes">60x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">371x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">349x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">340x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">340x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">330x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">92x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">92x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">92x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">92x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">92x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">92x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">20x</span>
<span class="cline-any cline-yes">360x</span>
<span class="cline-any cline-yes">290x</span>
<span class="cline-any cline-yes">290x</span>
<span class="cline-any cline-yes">1770x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">290x</span>
<span class="cline-any cline-yes">110x</span>
<span class="cline-any cline-yes">110x</span>
<span class="cline-any cline-yes">110x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">564x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">110x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { GridLengths } from '../constants/constants';
import { ExplodedBoxes, GridArray, CurrentColor, AppStore } from '../types/types';
import { boxPositionN } from './gameLogic';
&nbsp;
function floodFill(store: AppStore, position: { y: number, x: number }) {
    const copyGrid: GridArray = JSON.parse(JSON.stringify(store.gridArray));
    const current: CurrentColor = store.gridArray[position.y][position.x];
&nbsp;
    const explodedBoxes: ExplodedBoxes = [];
&nbsp;
    fill(copyGrid, position.y, position.x, current, explodedBoxes);
&nbsp;
    <span class="missing-if-branch" title="else path not taken" >E</span>if (explodedBoxes.length &gt; 2) {
        explodedBoxes.sort((a, b) =&gt; a.y - b.y);
        explodedBoxes.forEach((explodedBox) =&gt; {
            store.blinkingBoxesN.push(boxPositionN(explodedBox.y, explodedBox.x));
            store.checkBoxPositions.push({ y: explodedBox.y, x: explodedBox.x });
            store.explodedBoxes.push({ y: explodedBox.y, x: explodedBox.x });
        });
    }
};
&nbsp;
function fill(copyGrid: GridArray, y: number, x: number, current: CurrentColor, explodedBoxes: ExplodedBoxes): void {
    // If row is less than 0 - bottom case
    if (y &lt; 0) return;
&nbsp;
    // If column is less than 0 - bottom case
    if (x &lt; 0) return;
&nbsp;
    // If row is greater than copyGrid length - bottom case
    <span class="missing-if-branch" title="if path not taken" >I</span>if (y &gt; GridLengths.RowLength - 1) <span class="cstat-no" title="statement not covered" >return;</span>
&nbsp;
    // If column is greater than copyGrid length - bottom case
    if (x &gt; GridLengths.ColumnsLength - 1) return;
&nbsp;
    // If this position does not equal the color or is null - bottom case
    if (copyGrid[y][x] !== current || copyGrid[y][x] === null) return;
&nbsp;
    copyGrid[y][x] = null;
&nbsp;
    explodedBoxes.push({ y: y, x: x });
&nbsp;
    // Fill in all four directions
    // Fill Prev row
    fill(copyGrid, y - 1, x, current, explodedBoxes);
&nbsp;
    // Fill next row
    fill(copyGrid, y + 1, x, current, explodedBoxes);
&nbsp;
    // Fill prev col
    fill(copyGrid, y, x - 1, current, explodedBoxes);
&nbsp;
    // Fill next col
    fill(copyGrid, y, x + 1, current, explodedBoxes);
}
&nbsp;
// After flood fill destroys boxes, fill the null spaces with boxes above (if there are any)
function fillEmptyGridSpaces(store: AppStore): void {
    const changedBoxes: ExplodedBoxes = [];
&nbsp;
&nbsp;
    // SCORE IS ADDED HERE!
    store.explodedBoxes.forEach(<span class="fstat-no" title="function not covered" >po</span>sition =&gt; {
<span class="cstat-no" title="statement not covered" >        store.gridArray[position.y][position.x] = null;</span>
<span class="cstat-no" title="statement not covered" >        store.score += 10;</span>
    });
&nbsp;
    for (let x = 0; x &lt; GridLengths.ColumnsLength; x++) {
        for (let y = 0; y &lt; GridLengths.RowLength - 2; y++) {
            if (store.gridArray[y][x] === null) {
                let nextNonNull = y + 1;
                while (nextNonNull &lt; GridLengths.RowLength &amp;&amp; store.gridArray[nextNonNull][x] === null) {
                    nextNonNull++;
                }
                if (nextNonNull &lt; GridLengths.RowLength - 1) {
                    store.gridArray[y][x] = store.gridArray[nextNonNull][x];
                    store.gridArray[nextNonNull][x] = null;
                    changedBoxes.push({ y: y, x: x });
                }
            }
        }
    }
&nbsp;
    store.explodedBoxes = [];
&nbsp;
    changedBoxes.sort((a, b) =&gt; a.y - b.y);
&nbsp;
    return changedBoxes.forEach((position) =&gt; {
        store.checkBoxPositions.push(position);
    });
}
&nbsp;
export { floodFill, fill, fillEmptyGridSpaces };</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2024-05-08T15:43:17.384Z
            </div>
        <script src="../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../sorter.js"></script>
        <script src="../block-navigation.js"></script>
    </body>
</html>
    