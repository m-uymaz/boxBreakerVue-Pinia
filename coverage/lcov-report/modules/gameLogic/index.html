
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for modules/gameLogic</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> modules/gameLogic</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">95.6% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>87/91</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">89.28% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>25/28</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">95% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>19/20</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">96.15% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>75/78</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="boxPositionN.ts"><a href="boxPositionN.ts.html">boxPositionN.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

<tr>
	<td class="file high" data-value="findRowColIndex.ts"><a href="findRowColIndex.ts.html">findRowColIndex.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	</tr>

<tr>
	<td class="file high" data-value="floodFillFuncs.ts"><a href="floodFillFuncs.ts.html">floodFillFuncs.ts</a></td>
	<td data-value="93.75" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 93%"></div><div class="cover-empty" style="width: 7%"></div></div>
	</td>
	<td data-value="93.75" class="pct high">93.75%</td>
	<td data-value="48" class="abs high">45/48</td>
	<td data-value="90" class="pct high">90%</td>
	<td data-value="20" class="abs high">18/20</td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	<td data-value="94.87" class="pct high">94.87%</td>
	<td data-value="39" class="abs high">37/39</td>
	</tr>

<tr>
	<td class="file high" data-value="generateInitialGrid.ts"><a href="generateInitialGrid.ts.html">generateInitialGrid.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

<tr>
	<td class="file high" data-value="generateNewLine.ts"><a href="generateNewLine.ts.html">generateNewLine.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	</tr>

<tr>
	<td class="file empty" data-value="index.ts"><a href="index.ts.html">index.ts</a></td>
	<td data-value="0" class="pic empty">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file high" data-value="randomColor.ts"><a href="randomColor.ts.html">randomColor.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file high" data-value="timeouts.ts"><a href="timeouts.ts.html">timeouts.ts</a></td>
	<td data-value="93.33" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 93%"></div><div class="cover-empty" style="width: 7%"></div></div>
	</td>
	<td data-value="93.33" class="pct high">93.33%</td>
	<td data-value="15" class="abs high">14/15</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="93.33" class="pct high">93.33%</td>
	<td data-value="15" class="abs high">14/15</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2024-05-08T15:17:21.063Z
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    