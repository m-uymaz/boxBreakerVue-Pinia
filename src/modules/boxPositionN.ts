export const boxPositionN = (y: number, x: number) => {
  return y * 10 + (x + 1)
}
