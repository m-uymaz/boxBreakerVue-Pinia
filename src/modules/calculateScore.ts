const calculateScore = (score: number): string => {
    const zerosArr = ['0', '0', '0', '0', '0', '0'];
    const appScoreString: string = score.toString();

    const scoreToDisplayArr: string[] = zerosArr.slice(0, zerosArr.length - appScoreString.length);

    const scoreString: string = scoreToDisplayArr.join('') + appScoreString;

    return scoreString;
}

export default calculateScore;