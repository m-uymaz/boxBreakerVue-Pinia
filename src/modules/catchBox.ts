import { GridRowIndeces } from '../constants/constants.js';
import { AppStateInterface } from '../types/types.js';

// Catch the first available box
const catchBox = (store: AppStateInterface): void => {
    for (let y = GridRowIndeces.NextToLast; 0 <= y; y--) {
        if (store.explodedBoxes.filter(obj => obj.y === y && obj.x === store.arrowIndex).length) break;

        if (store.gridArray[y][store.arrowIndex] === null) {
            continue;
        } else {
            store.coughtBox = store.gridArray[y][store.arrowIndex];
            store.gridArray[y][store.arrowIndex] = null;
            store.coughtBoxFrom = { y: y, x: store.arrowIndex }
            break;
        }
    }
    // console.clear();
    // console.table(gridArray);
}

export default catchBox;