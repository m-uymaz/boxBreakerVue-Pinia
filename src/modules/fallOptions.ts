import { AppStateInterface } from "../types/types";

import moveDown from "./moveDown";

const fallOn = (store: AppStateInterface) => {
    if (store.fall) return;
    store.fall = true;
    store.interval = setInterval(() => {
        if (
            !store.fall
            ||
            store.timeouts.explodeTimeout
            ||
            store.timeouts.rerenderTimeout
        ) return;

        if (store.gameOverState && store.interval) {
            clearInterval(store.interval);
            store.interval = null;
        }

        if (store.countMilliseconds === 2500) {
            moveDown(store);
            store.countMilliseconds = 0;
        }

        store.countMilliseconds += 100;
    }, 100);
}

const fallOff = (store: AppStateInterface) => {
    if (!store.fall) return;
    store.fall = false;
    if (store.interval) {
        clearInterval(store.interval);
        store.interval = null;
        store.countMilliseconds = 0;
    }
}

export {fallOn, fallOff}

