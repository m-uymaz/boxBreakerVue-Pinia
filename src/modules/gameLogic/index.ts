export * from '../boxPositionN';
export * from '../findRowColIndex';
export * from '../floodFillFuncs';
export * from '../timeouts';
export * from '../randomColor';
export * from '../generateInitialGrid';
export * from '../generateNewLine';
export * from '../gameOverBoxDanceFuncs';