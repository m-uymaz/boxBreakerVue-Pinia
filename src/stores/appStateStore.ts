import { defineStore } from "pinia";

import { generateInitialGrid } from "../modules/gridArray";
import { AppStateInterface } from "../types/types";

export const useAppStateStore = defineStore('appStateStore', {
    state: (): AppStateInterface => ({
        gridArray: generateInitialGrid(),
        timeouts: {
            explodeTimeout: null,
            rerenderTimeout: null,
        },
        checkBoxPositions: [],
        explodedBoxes: [],
        blinkingBoxesN: [],
        explodingBoxesN: [],
        highestPositionY: 1,
        arrowIndex: 5,
        coughtBox: null,
        coughtBoxFrom: null,
        thrownBox: null,
        gameOverState: false,
        fall: false,
        interval: null,
        countMilliseconds: 0,
        score: 0,
    })
})