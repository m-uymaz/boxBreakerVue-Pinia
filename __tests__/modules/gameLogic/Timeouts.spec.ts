import { setActivePinia, createPinia } from "pinia";
import { explodeDelay, clearPrevTimeouts } from "../../../src/modules/gameLogic";
import { useAppStateStore } from "../../../src/stores/appStateStore";

describe("Timeouts", () => {
    beforeEach(() => {
        // creates a fresh pinia and makes it active
        // so it's automatically picked up by any useStore() call
        // without having to pass it to it: `useStore(pinia)`
    setActivePinia(createPinia());
    });

    Object.defineProperty(global, 'performance', {
        writable: true,
    });

    jest.useFakeTimers();
    jest.spyOn(global, "setTimeout");

    test('explodingBoxeN takes arr values of blinkingBoxesN and blinkingBoxesN is left with no values in arr', () => {
        const store = useAppStateStore();
        const seedArrBlinkingBoxes = [11, 12, 13, 14, 15, 16, 17, 18, 19, 20];  
        const timeoutFunc = jest.fn(() => explodeDelay(store, 1000));
    
        store.blinkingBoxesN = [...seedArrBlinkingBoxes];
        expect(store.blinkingBoxesN).toEqual(seedArrBlinkingBoxes);
        expect(timeoutFunc).not.toHaveBeenCalled();  

        timeoutFunc();
        jest.advanceTimersByTime(500);
    
        expect(store.blinkingBoxesN).toEqual(seedArrBlinkingBoxes);
        expect(store.explodingBoxesN).toEqual([]);

        jest.advanceTimersByTime(500);
    
        expect(timeoutFunc).toHaveBeenCalledTimes(1);
        expect(store.blinkingBoxesN).toEqual([]);
        expect(store.explodingBoxesN).toEqual(seedArrBlinkingBoxes);
    });
    
    test('clearing active timeouts', () => {
        const store = useAppStateStore();
        const timeoutFunc = jest.fn(() => explodeDelay(store, 1000));

        timeoutFunc();
        jest.advanceTimersByTime(500);

        expect(typeof store.timeouts.explodeTimeout).toBe('object');

        clearPrevTimeouts(store);

        expect(store.timeouts.explodeTimeout).toBeNull();
    });
});
