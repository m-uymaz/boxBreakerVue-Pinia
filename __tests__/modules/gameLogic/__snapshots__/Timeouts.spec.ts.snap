// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeouts explodeDelay() promise should catch err and reject 1`] = `"Time given is too big"`;
